testing/fuzzing/snmp_print_var_fuzzer.c

```diff
@@ -0,0 +1,271 @@
+            variable.val.string = (u_char *) s2;
+            variable.val_len = strlen(s2);
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 5:{
+            variable.type = ASN_GAUGE;
+            long            value = 3;
+            variable.val.integer = &value;
+            variable.val_len = 4;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 6:{
+            variable.type = ASN_COUNTER64;
+            struct counter64 c64;
+            c64.low = 0;
+            c64.high = 1;
+            variable.val.counter64 = &c64;
+            variable.val_len = 1;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 7:{
+            variable.type = ASN_TIMETICKS;
+            long            value = 3;
+            variable.val.integer = &value;
+            variable.val_len = 4;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 8:{
+            variable.type = ASN_OBJECT_ID;
+            variable.val.objid = (oid *) s2;
+            variable.val_len = strlen(s2);
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 9:{
+            variable.type = ASN_COUNTER;
+            long            value = 3;
+            variable.val.integer = &value;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 10:{
+            variable.type = ASN_UINTEGER;
+            long            value = 3;
+            variable.val.integer = &value;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 11:{
+            variable.type = ASN_OPAQUE_DOUBLE;
+            double          value = 3.3;
+            variable.val.doubleVal = &value;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 12:{
+            variable.type = ASN_OPAQUE_FLOAT;
+            float           fVal = 3.3;
+            variable.val.floatVal = &fVal;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    case 13:{
+            variable.type = ASN_NULL;
+            variable.next_variable = NULL;
+            print_variable(objid, objidlen, &variable);
+        }
+        break;
+    default:
+        break;
+    }
+
+    /*
+     * Fuzz snprint_* functions
+     */
+    variable.type = ASN_INTEGER;
+    long            value1 = 3;
+    long           *value1_ptr = &value1;
+    variable.val.integer = (long *) &value1_ptr;
```

Comment:
Shouldn't this be changed into `variable.val.integer = &value1;` since the variable type is `ASN_INTEGER`?

Reply:
Fixed! I did this essentially to accommodate for this line: https://github.com/net-snmp/net-snmp/blob/master/snmplib/mib.c#L1276 
Which formats a string based on `hints`. Since `hints` was fuzzed data, it could use a `%s` or so on the dereference `variable.val.integer`. However, now, instead I limit what `hints` can be.